	Bugfix of:  http://gcc.gnu.org/bugzilla/show_bug.cgi?id=55352
	gfortran-4.7.2 -Wall gives a warning about unused module variables when those
	variables are only used in a namelist

	Reported by: AstroFloyd@gmail.com
	Patch by: janus@gcc.gnu.org

Index: gcc/fortran/trans-decl.c
===================================================================
--- gcc/fortran/trans-decl.c    (revision 193567)
+++ gcc/fortran/trans-decl.c    (working copy)
@@ -4589,23 +4589,26 @@ generate_local_decl (gfc_symbol * sym)
     }

       /* Warn for unused variables, but not if they're inside a common
-     block, a namelist, or are use-associated.  */
+     block or a namelist.  */
       else if (warn_unused_variable
-           && !(sym->attr.in_common || sym->attr.use_assoc || sym->mark
-            || sym->attr.in_namelist))
+           && !(sym->attr.in_common || sym->mark || sym->attr.in_namelist))
     {
-      gfc_warning ("Unused variable '%s' declared at %L", sym->name,
-               &sym->declared_at);
-      if (sym->backend_decl != NULL_TREE)
-        TREE_NO_WARNING(sym->backend_decl) = 1;
+      if (sym->attr.use_only)
+        {
+          gfc_warning ("Unused module variable '%s' which has been "
+               "explicitly imported at %L", sym->name,
+               &sym->declared_at);
+          if (sym->backend_decl != NULL_TREE)
+        TREE_NO_WARNING(sym->backend_decl) = 1;
+        }
+      else if (!sym->attr.use_assoc)
+        {
+          gfc_warning ("Unused variable '%s' declared at %L",
+               sym->name, &sym->declared_at);
+          if (sym->backend_decl != NULL_TREE)
+        TREE_NO_WARNING(sym->backend_decl) = 1;
+        }
     }
-      else if (warn_unused_variable && sym->attr.use_only)
-    {
-      gfc_warning ("Unused module variable '%s' which has been explicitly "
-               "imported at %L", sym->name, &sym->declared_at);
-      if (sym->backend_decl != NULL_TREE)
-        TREE_NO_WARNING(sym->backend_decl) = 1;
-    }

       /* For variable length CHARACTER parameters, the PARM_DECL already
      references the length variable, so force gfc_get_symbol_decl
